### PR TITLE
Update usages of `KeyCombination.ReadableString()` to use `ReadableKeyCombinationProvider`

### PR DESCRIPTION
**Pull request by Susko3**
Depends on:
- [x] framework bump

osu!-side PR for https://github.com/ppy/osu-framework/pull/4834.

Not 100% on the font change in cc286f165d9e7a693377acd8f5347d7d13e5bb31.

It's unfortunate that the sizes don't match. Maybe it would look fine with `Numeric` + `Default` for fallback if sizes had matched.
The default font does look to be a bit off center, like it's a bit too much down. The `Ž` looks centered, so at least there's that.

`OsuFont.Numeric.With(size: 10)`:

image

`OsuFont.Default`:

image

### CODE REVIEW

**Comment by peppy** [comment]
Not sure about the font change either. I think might be fine to leave it as it is and wait for better fallback logic? The case you show in the screenshots is very edge-casey.

**Comment by bdach** [comment]
Yeah font fallback is an issue in more contexts, not only here. Even torus -> noto fallback looks bad. I'd probably leave numeric/venera there because it looks better alignment-wise due to all letters being uppercased.

File: osu.Game/Overlays/Settings/Sections/Input/KeyBindingRow.cs [hunk 1]
```diff
@@ -470,12 +477,19 @@ public KeyButton(RealmKeyBinding keyBinding)
...
                     },
                     new HoverSounds()
                 };
             }
 
+            protected override void LoadComplete()
+            {
+                base.LoadComplete();
+
+                readableKeyCombinationProvider.KeymapChanged += updateKeyCombinationText;
```
Comment (peppy) [inline osu.Game/Overlays/Settings/Sections/Input/KeyBindingRow.cs:489]: 99.99% sure this is not safe as the events are going to be fired from the input thread. feels like something that should be handled at a framework level potentially? probably only saved by the fact that updating `Text` on non-update thread is not safe but also doesn't throw..?

**Review by peppy — APPROVED** [review]

**Comment by bdach** [comment]
Looks fine here. Also tested with keymap changes and works as expected.

Unsure if you want https://github.com/ppy/osu-framework/pull/4869 to go with that but that can always happen later, I guess?

**Review by bdach — APPROVED** [review]

**Pull request merged**